packages/docusaurus/src/server/routes.ts

```diff
@@ -82,7 +83,12 @@ export function genChunkName(
       const name = str === '/' ? 'index' : docuHash(str);
       chunkName = prefix ? `${prefix}---${name}` : name;
     }
+    const seenCount = (chunkNameCount.get(chunkName) ?? 0) + 1;
+    if (seenCount > 1) {
+      chunkName += seenCount;
+    }
```

Comment:
Ah yes so it was 1 chance out of only 4096 to get a collision (`simpleHash(modulePath, 3)` returning 3 hex characters).

Not sure how much you want to refactor/clean this, I'll just add a few notes (feel free to ignore of course, I don't even know how implicated you are in the project :slightly_smiling_face: ):
- if we go for a counter, should we keep the hash? It seems unnecessary to have both.
- it seems unnecessary to do a `docuHash` after having done a `simpleHash`: if two `simpleHash`s produce the same output (`c99` in our case), then the two `docuHash`s will produce the same output as well (`485`).
- `docuHash` tests if `str` equals `/`, so we don't need to do `str === '/' ? 'index' : docuHash(str)` here.
- and a more general question: why don't we do `simpleHash(modulePath, 8)` in dev too?

---

For anyone discovering this PR, I've put together a small script to reproduce the issue we're having. Just run this with `ts-node`:

<details>

```ts
import { createHash } from "crypto";

export function md5Hash(str: string): string {
	return createHash("md5").update(str).digest("hex");
}

export function docuHash(str: string): string {
	if (str === "/") {
		return "index";
	}
	const shortHash = simpleHash(str, 3);
	const parsedPath = `${str}-${shortHash}`;
	console.log(`parsed path: ${parsedPath}`);
	return parsedPath;
}

export function simpleHash(str: string, length: number): string {
	return md5Hash(str).substring(0, length);
}

export function genChunkName(
	modulePath: string,
	prefix?: string,
	preferredName?: string,
	shortId: boolean = process.env.NODE_ENV === "production",
): string {
	let str = modulePath;

	if (preferredName) {
		const shortHash = simpleHash(modulePath, 3);

		console.log(`short hash of ${modulePath}: ${shortHash}`);

		str = `${preferredName}${shortHash}`;
	}

	const name = str === "/" ? "index" : docuHash(str);

	console.log(`docuHash of ${str}: ${name}`);

	let chunkName = prefix ? `${prefix}---${name}` : name;

	return chunkName;
}

console.log(
	genChunkName(
		"@site/blog/2022-11-18-bye-medium/index.mdx?truncated=true",
		"content",
		"/blog",
	),
);

console.log(
	genChunkName(
		"@site/blog/2019-10-05-react-nfc/index.mdx?truncated=true",
		"content",
		"/blog",
	),
);
```

Output:

```
short hash of @site/blog/2022-11-18-bye-medium/index.mdx?truncated=true: c99
parsed path: /blogc99-485
docuHash of /blogc99: /blogc99-485
content---/blogc99-485   <----- SAME RESULT HERE
short hash of @site/blog/2019-10-05-react-nfc/index.mdx?truncated=true: c99
parsed path: /blogc99-485
docuHash of /blogc99: /blogc99-485
content---/blogc99-485   <----- AND HERE
```

</details>

Reply:
> if we go for a counter, should we keep the hash?

The hash is also to make the IDs have a uniform and reasonably-short length. The ID is directly present in the client data, so any bytes we can shave off here is a net gain for bundle size.

I anticipate the counter to be very rarely used, and even if it is used, it would only add two bytes, which is fine.

> it seems unnecessary to do a `docuHash` after having done a `simpleHash`

`docuHash`'s name is confusing: hashing is probably the least that it does. It also normalizes the chunk name's capitalization and truncates the name, which makes sure the generated files can be properly emitted.

> `docuHash` tests if `str` equals `/`, so we don't need to do `str === '/' ? 'index' : docuHash(str)` here.

Thanks! This code has been there since forever (early in the conception of Docusaurus 2), so things probably have evolved since then.

> why don't we do `simpleHash(modulePath, 8)` in dev too?

For debuggability. The chunk name has some information of where it's coming from instead of an unreadable hash.

Reply:
Excellent. Thanks a lot for your answers, it all makes sense.